src/main/java/com/minecolonies/coremod/event/EventHandler.java

```diff
@@ -56,26 +70,114 @@ public void onDebugOverlay(final RenderGameOverlayEvent.Text event)
                 final WorldClient world = mc.world;
                 final EntityPlayerSP player = mc.player;
                 IColony colony = ColonyManager.getIColony(world, player.getPosition());
-                final double minDistance = ColonyManager.getMinimumDistanceBetweenTownHalls();
 
                 if (colony == null)
                 {
+                    if (!ColonyManager.isTooCloseToColony(world, player.getPosition()))
+                    {
+                        event.getLeft().add(LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.noCloseColony"));
+                        return;
+                    }
                     colony = ColonyManager.getClosestIColony(world, player.getPosition());
 
-                    if (colony == null || Math.sqrt(colony.getDistanceSquared(player.getPosition())) > 2 * minDistance)
+                    if (colony == null)
                     {
-                        event.getLeft().add(LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.noCloseColony"));
                         return;
                     }
 
                     event.getLeft().add(LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.nextColony",
-                      (int) Math.sqrt(colony.getDistanceSquared(player.getPosition())), minDistance));
+                            (int) Math.sqrt(colony.getDistanceSquared(player.getPosition())), ColonyManager.getMinimumDistanceBetweenTownHalls()));
                     return;
                 }
 
                 event.getLeft().add(colony.getName() + " : "
-                                      + LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.blocksFromCenter",
-                  (int) Math.sqrt(colony.getDistanceSquared(player.getPosition()))));
+                        + LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.blocksFromCenter",
+                        (int) Math.sqrt(colony.getDistanceSquared(player.getPosition()))));
+            }
+        }
+    }
+
+    /**
+     * Event called to attach capabilities.
+     * @param event the event.
+     */
+    @SubscribeEvent
+    public void onAttachingCapabilities(@NotNull final AttachCapabilitiesEvent<Chunk> event)
+    {
+        event.addCapability(new ResourceLocation(Constants.MOD_ID, "closeColony"), new ColonyTagCapabilityProvider());
+    }
+
+    //todo loading chunk public static void loadChunk(final World world, final int x, final int y, final int dim)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)